infra/image/dockerfile/c10s

```diff
@@ -0,0 +1,25 @@
+FROM quay.io/centos/centos:stream10-development
+ENV container=podman
+
+RUN rm -fv /var/cache/dnf/metadata_lock.pid; \
```

Comment:
uhmm,  I would combine commands using "&&" to ensure they run in a single layer and handle errors properly

what about:

```
# Install necessary packages and clean up
RUN rm -fv /var/cache/dnf/metadata_lock.pid && \
    dnf makecache && \
    dnf --assumeyes install \
        /usr/bin/python3 \
        /usr/bin/dnf-3 \
        sudo \
        bash \
        systemd \
        procps-ng \
        iproute && \
    rm -rf /var/cache/dnf/
```